### ISSUE TITLE
Crash when adding email block to ilert_connection

### ISSUE THREAD

**Issue by alexs77** [iLert/terraform-provider-ilert#10]
```hcl
data "ilert_escalation_policy" "default" {
  name = "Default"
}

resource "ilert_alert_source" "this" {
  name              = "Example Azure Alerts"
  integration_type  = "AZUREALERTS"
  escalation_policy = data.ilert_escalation_policy.default.id
}

resource "ilert_connector" "exampleemail" {
  name = "My email Connector"
  type = "email"
}

resource "ilert_connection" "testmail" {
  name = "testmail Connection"

  alert_source {
    id = ilert_alert_source.this.id
  }

  connector {
    id   = ilert_connector.exampleemail.id
    type = "email"
  }

  email {
    recipients = []
    subject    = ""
  }

  trigger_types = [
    "incident-acknowledged", "incident-assigned",
    "incident-auto-escalated", "incident-comment-added",
    "incident-created", "incident-raised", "incident-resolved"
  ]
}
```

When I try to `terraform apply` this, I constantly get a crash.

```sh
$ terraform version
Terraform v1.1.7
on linux_amd64
+ provider registry.terraform.io/hashicorp/azuread v2.18.0
+ provider registry.terraform.io/hashicorp/azurerm v2.96.0
+ provider registry.terraform.io/hashicorp/random v3.1.0
+ provider registry.terraform.io/ilert/ilert v1.4.7
```

Without the `email { … }` block, I'm able to `apply`, but there's then no recipient, subject and body defined. And I'd also always get a constant change.

```sh
$ terraform apply -auto-approve
…
Terraform will perform the following actions:

  # ilert_connection.testmail must be replaced
-/+ resource "ilert_connection" "testmail" {
      ~ created_at    = "2022-04-06T11:38:55.749Z" -> (known after apply)
      ~ id            = "…-71109a4f7290" -> (known after apply)
        name          = "testmail Connection"
      ~ updated_at    = "2022-04-06T11:38:55.749Z" -> (known after apply)
        # (2 unchanged attributes hidden)


      ~ connector {
          + id   = "…-18c387bf5543"
          + type = "email"
        }

      - email { # forces replacement
          - recipients = [] -> null
        }
        # (1 unchanged block hidden)
    }

Plan: 1 to add, 0 to change, 1 to destroy.
…
ilert_connection.testmail: Creation complete after 0s [id=…-70dafccee43c]

Apply complete! Resources: 1 added, 0 changed, 1 destroyed.
```

Terraform debug log from the following run ⇒ [_home_askwar_Code_ilert-test.log](https://github.com/iLert/terraform-provider-ilert/files/8426607/_home_askwar_Code_ilert-test.log)

```sh
$ env TF_LOG=DEBUG TF_LOG_PATH=/tmp/(pwd | tr / _).log terraform apply -auto-approve
ilert_connector.exampleemail: Refreshing state... [id=…-18c387bf5543]
ilert_alert_source.this: Refreshing state... [id=…278]
ilert_connection.testmail: Refreshing state... [id=…-70dafccee43c]

Terraform used the selected providers to generate the following execution plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform will perform the following actions:

  # ilert_connection.testmail will be updated in-place
  ~ resource "ilert_connection" "testmail" {
        id            = "…-70dafccee43c"
        name          = "testmail Connection"
        # (4 unchanged attributes hidden)


      ~ connector {
          + id   = "00a2249d-f73c-4e05-824b-18c387bf5543"
          + type = "email"
        }

        # (2 unchanged blocks hidden)
    }

Plan: 0 to add, 1 to change, 0 to destroy.
ilert_connection.testmail: Modifying... [id=…-70dafccee43c]
╷
│ Error: Plugin did not respond
│ 
│   with ilert_connection.testmail,
│   on iLert.tf line 16, in resource "ilert_connection" "testmail":
│   16: resource "ilert_connection" "testmail" {
│ 
│ The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ApplyResourceChange call. The plugin logs may contain more details.
╵

Stack trace from the terraform-provider-ilert_v1.4.7 plugin:

panic: interface conversion: interface {} is nil, not string

goroutine 54 [running]:
github.com/iLert/terraform-provider-ilert/ilert.buildConnection(0xc000124bf0)
	github.com/iLert/terraform-provider-ilert/ilert/resource_connection.go:969 +0x27c5
github.com/iLert/terraform-provider-ilert/ilert.resourceConnectionUpdate(0xc0002d5600, {0xeeefa0, 0xc0001a2cd8})
	github.com/iLert/terraform-provider-ilert/ilert/resource_connection.go:1227 +0x52
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).update(0xddbda0, {0x10db888, 0xc00004e300}, 0x24, {0xeeefa0, 0xc0001a2cd8})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.2.0/helper/schema/resource.go:292 +0x79
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0xc000226790, {0x10db888, 0xc00004e300}, 0xc00018a2a0, 0xc000570700, {0xeeefa0, 0xc0001a2cd8})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.2.0/helper/schema/resource.go:394 +0x6b0
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0xc0001a2648, {0x10db888, 0xc00004e300}, 0xc00071a640)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.2.0/helper/schema/grpc_provider.go:951 +0x9aa
github.com/hashicorp/terraform-plugin-go/tfprotov5/server.(*server).ApplyResourceChange(0xc00007c7c0, {0x10db930, 0xc000338210}, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/server/server.go:331 +0x6c
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0xebcc80, 0xc00007c7c0}, {0x10db930, 0xc000338210}, 0xc00039c060, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.1.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:380 +0x170
google.golang.org/grpc.(*Server).processUnaryRPC(0xc0002a4fc0, {0x10ec510, 0xc00055e180}, 0xc00012e200, 0xc000246330, 0x172b780, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1194 +0xc8f
google.golang.org/grpc.(*Server).handleStream(0xc0002a4fc0, {0x10ec510, 0xc00055e180}, 0xc00012e200, 0x0)
	google.golang.org/grpc@v1.32.0/server.go:1517 +0xa2a
google.golang.org/grpc.(*Server).serveStreams.func1.2()
	google.golang.org/grpc@v1.32.0/server.go:859 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
	google.golang.org/grpc@v1.32.0/server.go:857 +0x294

Error: The terraform-provider-ilert_v1.4.7 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by alexs77** [comment]
This also happens, when the `email { … }` contains data, like so:

```hcl
  email {
    recipients = ["foo@example.com"]
    subject    = "Alert"
  }
```

**Comment by yacut** [comment]
fixed in v1.5.1
